### ISSUE TITLE
Compile with `--prefix` doesn't work

### ISSUE THREAD

**Issue by scopatz** [kollokollo/X11Basic#7]
I am using conda to build and install xbasic.  However when I build with `./configure --prefix=/path/to/conda`, make fails to recognize this and will error out because [it can't find the readline headers](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=40140&view=logs&jobId=10bda42a-d420-5c28-a43f-bc4e66405e5c&taskId=bef775d0-a1a2-55a3-9721-17147029baf0&lineStart=716&lineEnd=719&colStart=1&colEnd=24):

```
do_gets.c:22:10: fatal error: readline/readline.h: No such file or directory
 #include <readline/readline.h>
          ^~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```

The issue where this occurs is at https://github.com/conda-forge/staged-recipes/pull/8509

**Comment by scopatz** [comment]
Note that I think this is because the custom CFLAGS variable in the environment prior to the build is not being respected by the build system here

**Comment by kollokollo** [comment]
I have not heard about conda-forge so far. Let me check....

It can well be, that the CFLAGS from the environment is ignored. So this would be a mater of the Makefile. At the moment I have no clue...

**Comment by kollokollo** [comment]
Try following:

In Makefile.in:line 66

> CFLAGS= @CFLAGS@ $(INC) $(DEF) $(OPT) $(REGS)

**Comment by kollokollo** [comment]
Maybe another line in the Makefile should be modified. I have attached a modified Makefile.in. I am not sure if this solves the issue. Need someone to test this. 
[a.zip](https://github.com/kollokollo/X11Basic/files/3260478/a.zip)

**Comment by scopatz** [comment]
Progress! Now it just fails to link against readline:

```
$BUILD_PREFIX/bin/x86_64-conda_cos6-linux-gnu-cc  -fPIC -shared -Wl,-Bsymbolic-functions  -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -I$PREFIX/include -fdebug-prefix-map=$SRC_DIR=/usr/local/src/conda/x11basic-1.26 -fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix  -DUSE_X11  -O3 -fno-omit-frame-pointer   -Wl,-soname,libx11basic.so.1.26 -o libx11basic.so.1.26 xbasic.c tools.c loadprg.c file.c io.c io_basic.c parser.c variablen.c svariablen.c array.c parameter.c fft.c mathematics.c memory.c runtime.c wort_sep.c ltext.c functions.c sfunctions.c afunctions.c kommandos.c gkommandos.c bitmap.c do_gets.c errortxt.c window.c sysVstuff.c aes.c graphics.c sound.c number.c decode.c virtual-machine.c type.c lodepng.c md5.c sha1.c -ldl -lm -lpthread -lutil -lreadline   
/home/scopatz/miniconda/conda-bld/x11basic_1559832336691/_build_env/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/bin/ld: cannot find -lreadline
collect2: error: ld returned 1 exit status
make: *** [Makefile:402: libx11basic.so.1.26] Error 1
```

**Comment by kollokollo** [comment]
OK, here is a next try
[a.zip](https://github.com/kollokollo/X11Basic/files/3264578/a.zip)

**Comment by scopatz** [comment]
I am now getting a lot further in the compilation, but now back to seeing:

```
do_gets.c:22:10: fatal error: readline/readline.h: No such file or directory
 #include <readline/readline.h>
          ^~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```

Here is [a gist with the build output](https://gist.github.com/scopatz/f7dd38ec18c12331365af7de3f1136a9)

**Comment by kollokollo** [comment]
Yes, thanks, this is useful information, however, as far as I can see, readline is not the only package missing, e.g. lapack, and even X11 (X.org). The corresponding -dev packages need to be installed before trying to compile,e.g.
sudo apt-get install libreadline-dev, Xorg-dev, liblapack-dev etc...
At least on a normal Ubuntu one would do this.

**Comment by kollokollo** [comment]
Are you sure, that the intermediate state (where it compiled but not linked) was really successfull? Maybe you have not redone the ./configure after placing a new Makefile.in. If you just type make (after having touched the Makefile.in), a prefix of /usr/bin is used.

**Comment by kollokollo** [comment]
Is this still an issue, or can this be closed?
